components/user_settings/display/manage_timezones.jsx

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedHTMLMessage, FormattedMessage} from 'react-intl';
+
+import {updateUser} from 'actions/user_actions.jsx';
+
+import SettingItemMax from 'components/setting_item_max.jsx';
+import {getTimezoneRegion, getBrowserTimezone} from 'utils/timezone';
+
+import SuggestionBox from 'components/suggestion/suggestion_box.jsx';
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+import TimezoneProvider from 'components/suggestion/timezone_provider.jsx';
+
+export default class ManageTimezones extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            useAutomaticTimezone: props.useAutomaticTimezone,
+            automaticTimezone: props.automaticTimezone,
+            manualTimezone: props.manualTimezone,
+            manualTimezoneInput: props.manualTimezone,
+            isSaving: false,
+        };
+    }
+
+    onChange = (e) => {
+        this.setState({manualTimezoneInput: e.target.value});
+    };
+
+    handleTimezoneSelected = (selected) => {
+        if (!selected) {
+            return;
+        }
+
+        this.setState({
+            manualTimezone: selected,
+            manualTimezoneInput: selected,
+        });
+    };
+
+    timezoneNotChanged = () => {
+        const {
+            useAutomaticTimezone,
+            automaticTimezone,
+            manualTimezone,
+        } = this.state;
+
+        const {
+            useAutomaticTimezone: oldUseAutomaticTimezone,
+            automaticTimezone: oldAutomaticTimezone,
+            manualTimezone: oldManualTimezone,
+        } = this.props;
+
+        return (
+            useAutomaticTimezone === oldUseAutomaticTimezone &&
+            automaticTimezone === oldAutomaticTimezone &&
+            manualTimezone === oldManualTimezone
+        );
+    };
+
+    changeTimezone = () => {
+        if (this.timezoneNotChanged()) {
+            this.props.updateSection('');
+            return;
+        }
+
+        this.submitUser();
+    };
+
+    submitUser = () => {
+        const {user} = this.props;
+        const {
+            useAutomaticTimezone,
+            automaticTimezone,
+            manualTimezone,
+        } = this.state;
+
+        const timezone = {
+            useAutomaticTimezone: useAutomaticTimezone.toString(),
```

Comment:
Why store this as a string version of the boolean? That seems like something that'll be easy to get mixed up